crates/bevy_picking/src/mesh_picking/ray_cast/intersections.rs

```diff
@@ -141,13 +144,15 @@ pub fn ray_mesh_intersection<I: TryInto<usize> + Clone + Copy>(
             };
             let triangle_index = Some(i);
             let tri_vertex_positions = &[Vec3::from(a), Vec3::from(b), Vec3::from(c)];
-            let tri_normals = vertex_normals.map(|normals| {
-                [
-                    Vec3::from(normals[i]),
-                    Vec3::from(normals[i + 1]),
-                    Vec3::from(normals[i + 2]),
-                ]
-            });
+            let tri_normals = vertex_normals
+                .map(|normals| -> Result<[Vec3; 3], ()> {
+                    let c = Vec3::from(*normals.get(i + 2).ok_or(())?);
+                    let b = Vec3::from(*normals.get(i + 1).ok_or(())?);
+                    let a = Vec3::from(*normals.get(i).ok_or(())?);
+                    Ok([a, b, c])
+                })
+                .transpose()
+                .ok()?;
```

Comment:
Same here. I'd double check the order is correct still:
```suggestion
            let tri_normals = vertex_normals.map(|normals| {
                Some([
                    Vec3::from(*normals.get(i)?),
                    Vec3::from(*normals.get(i + 1)?),
                    Vec3::from(*normals.get(i + 2)?),
                ])
            })?;
```

Reply:
see above